src/chrometrace.rs

```diff
@@ -23,9 +23,13 @@ pub fn write_chrometrace(target: &Path, profile_data: &[GpuTimerScopeResult]) ->
 }
 
 fn write_results_recursive(file: &mut File, result: &GpuTimerScopeResult, last: bool) -> std::io::Result<()> {
+    // hack to convert to integer
+    let tid_to_int = |tid| format!("{:?}", tid).replace("ThreadId(", "").replace(')', "").parse::<usize>().unwrap();
```

Comment:
Oof that's indeed a pretty nasty hack 😅 . But as you elaborated in the ticket and the PR description what can we possibly do :/

However, it would be good to document the why here and also the fact that the better solution via `as_u64` might be coming in the solution.
More importantly, we shouldn't crash here if for some reason the output changed and we fail to convert it to a usize. Instead we can fall back to something that is obviously a failed tid, i.e. `unwrap_or(std::u64::MAX)` maybe (let's go with u64 right away in expectation of that new method)